### PR TITLE
[MB-5431] - Add annotation for detect private key pre-commit hook

### PR DESCRIPTION
**Pull request by amitch23**
## Description

Accidentally merged [the previous PR](https://github.com/transcom/mymove/pull/5239), and unable to revert it. This implements the changes requested by @jneuner-mitre from https://dp3.atlassian.net/secure/RapidBoard.jspa?rapidView=38&modal=detail&selectedIssue=MB-5431

Based on discovery by @kimallen , all the keys except one case, are fake keys used for local development. Therefore, including an annotation indicating as much is the desired outcome.

The exception is in pkg/cli/auth.go which does not include an actual key- just the formatting for interpolating one.

This PR adds the annotation comments in the pre-commit config for the hook detect-private-key above the exclude block, specifying each file requiring exclusion.

## Reviewer Notes

It was not straightforward to include inline annotations for this pre-commit hook in the .ssh and .go files detected (see jira ticket for full list of files). As such, it seemed the most straightforward to use the exclude block in the pre-commit config, as it was intended to be used.

## Code Review Verification Steps

* [ ] If the change is risky, it has been tested in experimental before merging.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/wiki/Backend-Programming-Guide#querying-the-database-safely)
 have been satisfied.
* Any new migrations/schema changes:
  * [ ] Follow our guidelines for zero-downtime deploys (see [Zero-Downtime Deploys](https://github.com/transcom/mymove/wiki/migrate-the-database#zero-downtime-migrations))
  * [ ] Have been communicated to #g-database
  * [ ] Secure migrations have been tested following the instructions in our [docs](https://github.com/transcom/mymove/wiki/migrate-the-database#secure-migrations)
* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/secure/RapidBoard.jspa?rapidView=38&modal=detail&selectedIssue=MB-5431) for this change
* [this article](tbd) explains more about the approach used.
* [the old PR](https://github.com/transcom/mymove/pull/5239)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -19,15 +19,80 @@ repos:
...
+        #RA Summary: detect-private-key - Private key found for pkg/cli/auth.go$.
+        #RA: detect-private-key detected a private key, but the trigger in the CLI package is due to formatting the key via
+        #RA: string interpolation. That is, the key value itself is not included in this file.
+        #RA: The risk of a production key being introduced is mitigated by this detect-private-key hook failing, and
+        #RA: the PR requiring approval by a team designated to gate changes to this file, specifically attempts to exclude
+        #RA: new key additions.
+        #RA Developer Status: Mitigated
+        #RA Validator Status: {RA Accepted, Return to Developer, Known Issue, Mitigated, False Positive, Bad Practice}
+        #RA Validator: jneuner@mitre.org
+        #RA Modified Severity:
```
Comment (jneuner-mitre) [inline .pre-commit-config.yaml:76]:
```suggestion
        #RA Summary: detect-private-key - Private key found
        #RA: detect-private-key detected a private key in source control
        #RA: config/tls/devlocal-*.key files are used in devlocal testing environments only.
        #RA: pkg/server/testdata*.key files are used in __________ environments only.
        #RA: pkg/auth/authentication/auth_test.go contains a static key, used for unit tests in devlocal and pipeline only.
        #RA: pkg/cli/auth.go is a false positive; this code generates a keyfile at runtime, and its heuristics resemble an encoded key.
        #RA: .envrc.local.template is used in devlocal environments only.
        #RA: The risk of a production key being introduced is mitigated by this detect-private-key hook failing, and
        #RA: the PR requiring approval by a team designated to gate changes to this file, specifically attempts to exclude
        #RA: new key additions.
        #RA Developer Status: Mitigated
        #RA Validator Status: {RA Accepted, Return to Developer, Known Issue, Mitigated, False Positive, Bad Practice}
        #RA Validator: jneuner@mitre.org
        #RA Modified Severity:
```

**Review by jneuner-mitre — COMMENTED** [review]
Ugh, there has to be a cleaner way to do this.
Keyfiles should be explicit when excluded, but I see that we should condense the annotations.
Thank you for your patience in demonstrating how my first suggestion was obnoxious.

**Comment by amitch23** [comment]
Thanks @jneuner-mitre ! I manually added the changes you suggested because I spelled "marine" incorrectly for a devlocal key file and the pre-test was throwing an error for detect-private-key (confirmation that it's working as intended!)

**Review by jneuner-mitre — COMMENTED** [review]
The RA looks good, but not sure what's going on in server_test.

**Comment by amitch23** [comment]
@jneuner-mitre it was an intermittent failure in server_test that I reran and passed. Should be good to approve now :)

**Review by lynzt — COMMENTED** [review]
LGTM... leaving approval to Jarod

**Review by jneuner-mitre — APPROVED** [review]
🚀

**Review by lynzt — APPROVED** [review]
based on Jarod's feedback, LGTM 🚢

**Pull request merged**